lib/solveEquation/stepThrough.js

```diff
@@ -107,6 +157,69 @@ function stepThrough(leftNode, rightNode, comparator, debug=false) {
   return steps;
 }
 
+/*
+  Helper function that returns the roots and symbol of an input equation
+  that has roots to be found.
+  For every factor on the left hand side, solve a new equation that is factor = 0
+  for the symbol.
+  TODO: handle multiple variable solutions e.g (x + 2) (y + 2) = 0
+*/
+
+function getSolutionsAndSymbol (equation) {
+  const left = equation.leftNode;
+  const rightNode = Node.Creator.constant(0);
```

Comment:
edit: I came back and kinda see what you're doing - but can you define this right before you use it instead of here?